devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java

```diff
@@ -56,13 +56,21 @@
     private static final String MAIN_RESOURCES_OUTPUT = "build/resources/main";
     private static final String CLASSES_OUTPUT = "build/classes";
     private static final String DEPLOYMENT_CONFIGURATION = "deploymentConfiguration";
+    private static final String CLASSPATH_CONFIGURATION = "classpathConfiguration";
 
     private static Configuration classpathConfig(Project project, LaunchMode mode) {
         if (LaunchMode.TEST.equals(mode)) {
             return project.getConfigurations().getByName(JavaPlugin.TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME);
         }
         if (LaunchMode.DEVELOPMENT.equals(mode)) {
-            return project.getConfigurations().getByName(QuarkusPlugin.DEV_MODE_CONFIGURATION_NAME);
+            Configuration classpathConfiguration = project.getConfigurations().findByName(CLASSPATH_CONFIGURATION);
+            if (classpathConfiguration != null) {
+                project.getConfigurations().remove(classpathConfiguration);
+            }
+
+            return project.getConfigurations().create(CLASSPATH_CONFIGURATION).extendsFrom(
+                    project.getConfigurations().getByName(QuarkusPlugin.DEV_MODE_CONFIGURATION_NAME),
+                    project.getConfigurations().getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME));
```

Comment:
Just to confirm, this will include compile-only, runtime and quarkusDev scopes?

Reply:
I will update it has it effectively should.